### PR TITLE
Add old readiness probe related ENVs 

### PR DESCRIPTION
**Pull request by kvalliyurnatt**
We are seeing issues with upgrading to 2.14.0 when there are overrides present for the readiness probe, adding the ENVs that have the credentials for the old readiness probe to run successfully should fix that issues
fixes: https://github.com/elastic/cloud-on-k8s/issues/8006

### CODE REVIEW

**Comment by naemono** [comment]
Tested with the following manifest successfully
```
apiVersion: elasticsearch.k8s.elastic.co/v1
kind: Elasticsearch
metadata:
  name: elasticsearch-sample
spec:
  version: 8.15.0
  nodeSets:
  - name: master
    config:
      node.roles: ["master"]
      node.store.allow_mmap: false
    podTemplate:
      metadata:
        labels:
          foo: bar
      spec:
        containers:
        - name: elasticsearch
          readinessProbe:
            exec:
              command:
                - bash
                - -c
                - /mnt/elastic-internal/scripts/readiness-probe-script.sh
            failureThreshold: 3
            initialDelaySeconds: 10
            periodSeconds: 12
            successThreshold: 1
            timeoutSeconds: 12
          resources:
            limits:
              memory: 2Gi
              cpu: 1
    count: 3
  - name: data
    config:
      node.roles: ["data", "ingest", "ml"]
      node.store.allow_mmap: false
    podTemplate:
      metadata:
        labels:
          foo: bar
      spec:
        containers:
        - name: elasticsearch
          readinessProbe:
            exec:
              command:
                - bash
                - -c
                - /mnt/elastic-internal/scripts/readiness-probe-script.sh
            failureThreshold: 3
            initialDelaySeconds: 10
            periodSeconds: 12
            successThreshold: 1
            timeoutSeconds: 12
          resources:
            limits:
              memory: 2Gi
              cpu: 1
          env:
          - name: ES_JAVA_OPTS
            value: "-Xms1g -Xmx1g"
    count: 3
```

I see the proper env vars set on the sts:
```
            - name: PROBE_PASSWORD_PATH
              value: /mnt/elastic-internal/pod-mounted-users/elastic-internal-probe
            - name: PROBE_USERNAME
              value: elastic-internal-probe
            - name: READINESS_PROBE_PROTOCOL
              value: https
```

**Review by naemono — APPROVED** [review]
LGTM as a straight-forward fix

**Comment by ibalat** [comment]
Hi, I have upgraded 2.14 and it uses new readiness script checks 8080 port. But, we use **9200** port for elasticsearch container. How can I customize the script or is there another solution to solve the problem? I had updated `readiness-port-script.sh` with old `readiness-probe-script.sh` but it is not working because of missing env. At last status, pod is **NotReady**.

**readiness-port-script.sh**: `nc -z -v -w5 127.0.0.1 8080 `

Image

**Comment by pebrc** [comment]
@ibalat can you open a thread in https://discuss.elastic.co/c/orchestration/eck/79 and we can discuss there. Let's reserve this PR for the discussion of the actual PR

**Comment by kvalliyurnatt** [comment]
Closing this, as we don't want to reintroduce these ENVs again, we have updated the documentation on the new readiness probe

File: pkg/controller/elasticsearch/nodespec/defaults.go [hunk 1]
```diff
@@ -48,27 +47,22 @@ func DefaultEnvVars(v version.Version, httpCfg commonv1.HTTPConfig, headlessServ
...
 		// needed in elasticsearch.yml
 		{Name: settings.HeadlessServiceName, Value: headlessServiceName},
-	}
-	if v.LT(esv1.MinReadinessPortVersion) {
-		vars = []corev1.EnvVar{
-			{Name: settings.EnvProbePasswordPath, Value: path.Join(esvolume.PodMountedUsersSecretMountPath, user.ProbeUserName)},
-			{Name: settings.EnvProbeUsername, Value: user.ProbeUserName},
-			{Name: settings.EnvReadinessProbeProtocol, Value: httpCfg.Protocol()},
-			{Name: settings.HeadlessServiceName, Value: headlessServiceName},
+		{Name: settings.EnvProbePasswordPath, Value: path.Join(esvolume.PodMountedUsersSecretMountPath, user.ProbeUserName)},
```
Comment (pebrc) [inline pkg/controller/elasticsearch/nodespec/defaults.go:50]: Let's maybe add a comment here that they are only added for BWC and considered deprecated to be removed in a future release.

**Review by pebrc — APPROVED** [review]

**Comment by kvalliyurnatt** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.15|[](https://github.com/elastic/cloud-on-k8s/pull/8155)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**